src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -1185,16 +1240,19 @@ private IndexWriter CreateIndexWriter()
             // ensure that it's unlocked here in that case.
             try
             {
-                if (IndexWriter.IsLocked(dir))
+                if (IndexWriter.IsLocked(dir) && (!(dir is ExamineDirectory) ||
```

Comment:
If we are going to continue using `ExamineDirectory` to check for things like readonly we should encapsulate an `IsReadOnly` property instead of having to check for `is ExamineDirectory` everywhere